### ISSUE TITLE
Installation error on Asus Tuf Gaming A15/A17

### ISSUE THREAD

**Issue by Qudratillo** [pop-os/pop#1199]
**Distribution (run `cat /etc/os-release`):**

 Pop!_OS 20.04

**Issue/Bug Description:**

Hi. I am trying to install Pop!_OS on my Asus Tuf Gaming A17 laptop (Ryzen 5 4600h, Geforce 1650ti). But I can't get installer running. At first it shows many lines and all disappear but "Started firmware update daemon". After that cursor keeps blinking and nothing happens. 
 I found another guy having exact issue with Tuf gaming A15. He even uploaded a video of process.
https://www.reddit.com/r/pop_os/comments/ggbrgr/black_screen_w_blinking_cursor_while_trying_to/?utm_source=amp&utm_medium=&utm_content=post_body

**Steps to reproduce (if you know):**

Just try to install on Tuf Gaming A17 laptop or watch this video
https://www.reddit.com/link/ggbrgr/video/mel9xerv7px41/player

**Expected behavior:**
Installation screen

**Other Notes:**

**Comment by Qudratillo** [comment]
Hello, @jackpot51 @mmstick @ids102 or anyone else. Can you help me?

**Comment by mmstick** [comment]
You'll have to ask ASUS for that support, or look for people with that exact model that have gotten it working on Linux. We can only realistically provide support for System76 systems.

**Comment by GavinRay97** [comment]
Holy moly, this was the install from hell. Most difficult install I've ever done in my life, hoping to share the journey here for any others who stumble on this:

The link from @DamnedWorkarounds is a good reference, one of three I found and patched together a working solution from:

1. https://www.reddit.com/r/pop_os/comments/hm5dd3/installing_pop_os_2004_on_asus_a15_fx506/
2. https://www.reddit.com/r/pop_os/comments/ggbrgr/black_screen_w_blinking_cursor_while_trying_to/fq698fy
3. (Right click the page and translate to English, it's in French) https://doc.ubuntu-fr.org/asus-a17-tuf766iu

His answer did not work for me -- trying to leave Secure Boot enabled and adding the `.efi` files to the trusted sources in BIOS was a no go.

The problem stems from nouveau drivers and that you need kernel +5.7 for this hardware to work properly (in my case, RTX 2060 and Ryzen 7 4800H).

This is roughly the process that worked for me (it was a lot of steps, trying my hardest to remember):
- I used the Intel, not Nvidia ISO because of comments online. Whether this was necessary or not I can't say for certain
- Press F2 during boot to get to BIOS, once in BIOS press F7 to go to advanced settings. Disable "Fast Boot" and "Secure Boot" from the menus (they are in separate tabs, Secure Boot is under "security", Fast Boot I think it's the first tab)
- Press "Save Changes & Exit"
- Power on again, press F2 to get to BIOS, then F8 to open boot menu. Boot from the USB with Pop ISO
- When Pop is booting, press (space? e?) to open the GRUB settings. It pops up the mini-Emacs editor. Find the line that starts with `linux`, and add at the end `nouveau.modeset=0`. Then press `F10` or `Ctrl + X`.
  - https://pop-planet.info/wiki/index.php?title=Kernel_parameters&oldid=387#GRUB
  - The screen looks roughly like this: ![image](https://user-images.githubusercontent.com/26604994/91627144-4b356700-e983-11ea-8f3e-e88bfb523cc6.png)
- Okay now the broken nouveau drivers are disabled. Reboot, and this time you should be able to load the graphical installer (the screen might go dim for 20-30 seconds and look like it's stuck, give it maybe a full minute before assuming it's broken again now)
- Run the standard installation. When it's finished, it'll break again on next reboot.
- Now repeat the process of disabling `nouveau.modeset=0` on this boot so that you can create a user account. You'll need to press `space` while booting up, then `e` to edit the params, and tack on the `nouveau.modeset=0` at the end then press `enter`.
- Should pop you into a standard session (maybe again 20-30 second wait on screen). Here you will finish creating a user account. If you can somehow do this BEFORE this point, during the initial install, save yourself headache and do it then. I don't know if this is possible.
- If it reboots again, use `Ctrl + Alt + F2` to switch TTY's on the hung boot process, and it should pop you into a shell. Log in using the username + password you created. If not, just open a shell graphically.
- Install the 5.8 mainline kernel:
  - ```sh
	wget https://raw.githubusercontent.com/pimlie/ubuntu-mainline-kernel.sh/master/ubuntu-mainline-kernel.sh
    chmod +x ubuntu-mainline-kernel.sh
    sudo mv ubuntu-mainline-kernel.sh /usr/local/bin/
    sudo ubuntu-mainline-kernel.sh -i
    ```
 - This should prompt you to install `5.8`, press yes. Wait for it to finish.
- Now at this point, you can reboot and everything should load. Check your kernel version with `uname -a`. The last bit is that the GPU is using Renoir integrated.
- Install the Nvidia 450 drivers.
  ```sh
  sudo add-apt-repository ppa:graphics-drivers/ppa
  sudo apt update
  sudo apt install nvidia-450
  ```
- Reboot
- Run `nvidia-smi`, or `sudo apt install neofetch` then `neofetch` to check everything is good.

**Comment by alex-pek10** [comment]
@GavinRay97 Thank you so much! I have been fighting with not just Pop_OS but pretty much EVERY distribution and your steps finally worked and gave me a functional system. THANK YOU!

**Comment by rahultrivedi180** [comment]
worked for me @GavinRay97 

Installed ubuntu 20.04.1. Thank you so much

**Comment by ThomasFlooksBMG** [comment]
DUUUUUUUDDDDEEEEEEE!!!! 
You are a legend for posting this solution.
It worked for me 👍

**Comment by alex-pek10** [comment]
With the FA506IU (TUF A15) I will still get a flicker in the bottom left corner once per second, however this can be fixed by switching to 60Hz refresh rate and then switching back to 144Hz. I think it has something to do with AMD Freesync, but that's just a theory. Minor annoyance and an easy fix when you notice it happening.

**Comment by chandanmalla** [comment]
Thanks GavinRay97  for the solution, it worked flawlessly on my system, but to add on, as you suggested to upgrade Kernel to 5.8, you need to install unofficial Nvidia driver version 450.

alex-pek10 , I thought there is problem in my display, i  was about to return it, thanks for your help :)

**Comment by GavinRay97** [comment]
Hey look, it helped people, wow awesome =D

**Comment by fonehackerx** [comment]
I am facing this issue as well. Will this be fixed in next pop os release? What pop os' general release date trend?

**Comment by keshavvinayak01** [comment]
> With the FA506IU (TUF A15) I will still get a flicker in the bottom left corner once per second, however this can be fixed by switching to 60Hz refresh rate and then switching back to 144Hz. I think it has something to do with AMD Freesync, but that's just a theory. Minor annoyance and an easy fix when you notice it happening.

Where you able to solve this issue? It's slightly annoying. Got the same TUF but with ryzen 5 + GTX 1650

**Comment by fonehackerx** [comment]
> > With the FA506IU (TUF A15) I will still get a flicker in the bottom left corner once per second, however this can be fixed by switching to 60Hz refresh rate and then switching back to 144Hz. I think it has something to do with AMD Freesync, but that's just a theory. Minor annoyance and an easy fix when you notice it happening.
> 
> Were you able to solve this issue? It's slightly annoying. Got the same TUF but with ryzen 5 + GTX 1650

I am no expert, but heard 20.10 (currently beta) version has got this fixed. Manjaro is currently stable on our laptop. People are using it. It's because of Kernel issues that Linux kernel 5.8 has fixed.

**Comment by THEBOSS619** [comment]
any issues for PopOS 20.10 on A15/17?

**Comment by rahultrivedi180** [comment]
> any issues for PopOS 20.10 on A15/17?

Works out of the box for PopOS 20.10. I am on it now

**Comment by THEBOSS619** [comment]
> > any issues for PopOS 20.10 on A15/17?
> 
> Works out of the box for PopOS 20.10. I am on it now

Oh thank you for your reply! you have just saved me alot of time now :)

**Comment by AnonymouX47** [comment]
Thanks so much for the comment... Just about the get the laptop, i was almost considering a different option after seeing different negative comments.

**Comment by vellankiindeevar** [comment]
is 21.04 working on A15 ?

**Comment by rahultrivedi180** [comment]
> is 21.04 working on A15 ?

Update the kernel if GUI doesn't start and it should work.

**Comment by AnonymouX47** [comment]
> is 21.04 working on A15 ?

Yes, currently using it.

**Comment by vellankiindeevar** [comment]
what about the asus control center thing ?

**Comment by landonh3x** [comment]
The way to fix the original issue of the ISO not booting from the USB stick into POP OS is to turn off "Secure Boot" from the boot menu. Place the USB partition in the top first on the boot order. Before updating this I recommend placing the RBG to a color you're going to like,  because that's one thing that you probably wouldn't be able to change with POP OS Nvidia LTS. I have an ASUS Tuf 15 Ryzen 7 1660ti and this works for me. I'm pretty sure I'll be switching to this as my main operation system with no dual boot since I'll be using the laptop for other things than gaming. Look to make sure once you switch all the hardware and BIOs has been updated from widows before the switch. If you decide to just update in POP OS im sure there are terminal commands that will help.

**Comment by jacobgkau** [comment]
My understanding is that this issue with Asus hardware was fixed by kernel 5.7. Pop!_OS is currently shipping a much newer version than that (5.19), so I'm closing this issue.

**Comment by landonh3x** [comment]
Thank you for your response and update!

For the most part it's about flawless on the Asus Tuf, Ryzen 7 1060ti

Really Glas this company made linux easy and practical.  I got my Linux
certifications using POP OS, will be a life time supporter.
